include/xamarin/sk_manageddrawable.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2015 Xamarin Inc.
+ * Copyright 2017 Microsoft Corporation. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#ifndef sk_manageddrawable_DEFINED
+#define sk_manageddrawable_DEFINED
+
+#include "sk_xamarin.h"
+
+#include "sk_types.h"
+
+SK_C_PLUS_PLUS_BEGIN_GUARD
+
+
+typedef struct sk_manageddrawable_t sk_manageddrawable_t;
+
+
+typedef void          (*sk_manageddrawable_draw_delegate)               (sk_manageddrawable_t* cmanagedDrawable, sk_canvas_t* ccanvas);
+typedef sk_rect_t     (*sk_manageddrawable_getBounds_delegate)          (sk_manageddrawable_t* cmanagedDrawable);
+typedef sk_picture_t* (*sk_manageddrawable_newPictureSnapshot_delegate) (sk_manageddrawable_t* cmanagedDrawable);
+
+
+SK_X_API sk_manageddrawable_t* sk_manageddrawable_new (void);
+SK_X_API void sk_manageddrawable_destroy (sk_manageddrawable_t*);
+SK_X_API void sk_manageddrawable_set_delegates (const sk_manageddrawable_draw_delegate pDraw,
+                                                const sk_manageddrawable_getBounds_delegate pGetBounds,
+                                                const sk_manageddrawable_newPictureSnapshot_delegate pNewPictureSnapshot);
+SK_X_API uint32_t sk_manageddrawable_get_generation_id (sk_manageddrawable_t*);
+SK_X_API sk_rect_t sk_manageddrawable_get_bounds (sk_manageddrawable_t*);
+SK_X_API void sk_manageddrawable_draw (sk_manageddrawable_t*, sk_canvas_t*, const sk_matrix_t*);
```

Comment:
I would split this up into 2 files - the `sk_drawable.h` in the `c` directory with the non-abstract methods, such as `draw` and `newPictureSnapshot`.

The reason for this is that `SkPicture::finishRecordingAsDrawable` does not actually return a managed drawable, and we need to be able to talk directly to the base `SkDrawable`.

This is sort of what happened with the managed streams.

There is a `SkStream` base that is abstract, but some APIs return that type. We need to be able to talk to that type without caring if it is a managed one or not. Then I created an abstract managed stream so that we can override the onX methods. Finally, I created a managed stream to wrap the .NET streams.

We should do something similar, except that we only want 2 types as we will never be instantiating a drawable from managed code without an inherited type. SO, we would have `SKDrawable` that is just gets instantiated from native alues, and then a `SKManagedDrawable` that the user would use.